app/test/unit/git/stash-test.ts

```diff
@@ -25,42 +27,119 @@ describe('git/stash', () => {
       const repo = await setupEmptyRepository()
       readme = path.join(repo.path, 'README.md')
       await FSE.writeFile(readme, '')
-      await stash(repo)
+      await stash(repo, 'master', null)
 
       expect(getDesktopStashEntries(repo)).rejects.toThrow()
     })
 
     it('returns all stash entries created by Desktop', async () => {
-      await generateTestStashEntries(repository)
+      await generateTestStashEntry(repository, 'master', false)
+      await generateTestStashEntry(repository, 'master', false)
+      await generateTestStashEntry(repository, 'master', true)
 
       const stashEntries = await getDesktopStashEntries(repository)
 
       expect(stashEntries).toHaveLength(1)
       expect(stashEntries[0].branchName).toBe('master')
     })
   })
+
+  describe('createStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
+
+    it('creates a stash entry', async () => {
+      await await FSE.appendFile(readme, 'just testing stuff')
+      const tipSha = await getTipSha(repository)
+
+      await createStashEntry(repository, 'master', tipSha)
+
+      const result = await GitProcess.exec(['stash', 'list'], repository.path)
+      const entries = result.stdout.trim().split('\n')
+      expect(entries).toHaveLength(1)
+    })
+  })
+
+  describe('getLastStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
+
+    it('returns null when no stash entries exist for branch', async () => {
+      await generateTestStashEntry(repository, 'some-other-branch', true)
+
+      const entry = await getLastStashEntry(repository, 'master')
+
+      expect(entry).toBeNull()
+    })
+
+    it('returns last entry made for branch', async () => {
+      const branchName = 'master'
+      await generateTestStashEntry(repository, branchName, true)
+      const lastEntry = await generateTestStashEntry(
+        repository,
+        branchName,
+        true
+      )
+
+      const actual = await getLastStashEntry(repository, branchName)
+
+      expect(actual!.stashSha).toBe(lastEntry)
+    })
+  })
 })
 
-async function stash(repository: Repository, message?: string) {
+async function getTipSha(repository: Repository) {
```

Comment:
💭 this helper function might be more useful here rather than rolling your own:

https://github.com/desktop/desktop/blob/385be4ddee002f6f04e776f86a2d73df0f00b19d/app/test/helpers/tip.ts#L5-L15